Update to .NET 10 SDK and Arcade 10

joperezr opened the pull request:
## Description

Upgrading to use the .NET 10 SDK as well as Arcade 10. This will be helpful for some signing things we are doing plus taking advantage of new features in the SDK.

cc: @radical 

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] No
- Did you add public API?
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] No

joperezr committed:
Update to .NET 10 SDK and Arcade 10

joperezr commented:
Setting as a draft as I'm queuing an internal build to see if official build works

joperezr committed:
Update workflows to instal 10 sdk as well

joperezr commented:
> Does this mean we could target `net10.0` in the Aspire.Cli when publishing for AOT?

If we wanted to, yes 😃

joperezr committed:
Use latest released preview of the sdk instead

joperezr committed:
Add all necessary runtimes for template tests.

joperezr committed:
Update tests for net10 now that targeting that from templates is supported.

joperezr committed:
Update the testSummary project so that it also tagets net10

radical commented:
We should also run an internal build, and a full azdo build with tests to validate this.

joperezr committed:
Adding sign info for js files

joperezr committed:
Add dotnet10 to the helix payload

davidfowl commented:
Let’s gooo!

radical commented:
New internal build - https://dev.azure.com/dnceng/internal/_build/results?buildId=2738295&view=results
New full azdo build - https://dev.azure.com/dnceng-public/public/_build/results?buildId=1079575&view=results

radical commented:
@joperezr would this work fine with VS/VS code?

radical committed:
Merge remote-tracking branch 'origin/main' into UpdateDotNet

radical committed:
Fix playground tests build on helix

`AzureFunctionsEndToEnd.AppHost.csproj : error NU1010: The following PackageReference items do not define a corresponding PackageVersion item: Aspire.Hosting.Azure.AppContainers. Projects using Central Package Management must declare PackageReference and PackageVersion items with matching names. For more information, visit https://aka.ms/nuget/cpm/gettingstarted `

radical committed:
Merge remote-tracking branch 'origin/main' into UpdateDotNet

radical committed:
Fix versions for current sdk to use 9.0 instead of 10.0, and revert the corresponding test change

radical committed:
[tests] Add properties to control filtering tests

radical committed:
fix md linter

radical committed:
Fix race on windows

radical reviewed (approved):
LGTM, thank you!!

radical commented:
@joperezr could you please review my additional changes to the PR before merging?

radical committed:
Merge remote-tracking branch 'origin/main' into UpdateDotNet

radical committed:
Merge remote-tracking branch 'origin/main' into UpdateDotNet

# Conflicts:
#	eng/clipack/Common.projitems

joperezr commented on eng/clipack/Common.projitems:
```diff
@@ -31,6 +31,14 @@
       <AdditionalProperties Include="SelfContained=true" />
+
+      <!-- Windows pdb aren't really needed any more. See https://github.com/dotnet/arcade/issues/15724
+          As the pdb file is in the same location for all the RIDs it causes arcade's `_DeployPortableSymbolsToSymStore`
```
Isn't the right fix here to use different publish locations for different rids?

joperezr commented on tests/Aspire.Templates.Tests/TemplateTestsBase.cs:
```diff
@@ -357,7 +357,7 @@ await CheckDashboardHasResourcesAsync(
             { templateName, extraArgs, TestSdk.Current, TestTargetFramework.Current, null },
             // Current SDK, Next TFM
-            { templateName, extraArgs, TestSdk.Current, TestTargetFramework.Next, "The current .NET SDK does not support targeting .NET 10.0" },
+            { templateName, extraArgs, TestSdk.Current, TestTargetFramework.Next, null },
```
Why not keep the current TFM actually be current (meaning 10.x) and then removing the expected error? (which is what I had) That way we are no validating that with a .NET 10 SDK you can build an aspire app targeting 10.

radical replied:
Can you elaborate on what is the missing case here, or the incorrect one?

radical commented on eng/Versions.props:
```diff
@@ -9,12 +9,12 @@
     <DotNetSdkPreviousVersionForTesting>8.0.406</DotNetSdkPreviousVersionForTesting>
     <!-- dotnet 10.0 versions for running tests - used for templates tests -->
-    <DotNetSdkNextVersionForTesting>10.0.100-preview.5.25277.114</DotNetSdkNextVersionForTesting>
+    <DotNetSdkCurrentVersionForTesting>9.0.200</DotNetSdkCurrentVersionForTesting>
```
These names are arbitrary really. We are following `Previous, Current, Next` to follow 3 active versions. `9.0` being `Current` works here allowing `Previous=8.0` and `Current=10.0`.

eerhardt replied:
But maybe that only makes sense if we are on `10` as well. So maybe having `Next` be 10 until we are version 10...

radical committed:
address review feedback from @ joperezr

joperezr commented on tools/GenerateTestSummary/GenerateTestSummary.csproj:
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net9.0</TargetFramework>
+    <TargetFramework>net10.0</TargetFramework>
```
We could, not sure if I'd block merging for this, so if this is the only remaining feedback I'd vote for doing this in the next PR
